### DISCUSSION TITLE
Error handling on the form api calls

### DISCUSSION THREAD

**Discussion by venkateshpullaganti** [manojadams/mui-forms#72] [Q&A] [answered]
Hi @manojadams 
We are trying to add an auto-filling city name based on the pin code entered, below is the city name field

` {
                    "name": "nominee_state_name",
                    "prop": "nominee",
                    "meta": {
                        "config": {
                            "type": "urlLoader",
                            "url": "https://some-pin-code-url/",
                            "queryParams": [
                                [
                                    "pincode",
                                    {
                                        "type": "fieldValue",
                                        "ref": "nominee_pincode"
                                    }
                                ]
                            ],
                            "valueKey": "state#name",
                            "labelKey": "state#name"
                        },
                        "displayName": "State",
                        "displayType": "select",
                        "options":[],
                        "displayProps": {
                            "xs": 12,
                            "md": 6
                        },
                        "validation": {
                            "required": true
                        },
                        "value": "",
                        "dependencies": {
                            "exists": {
                                "ref": "nominee_address_same_as_proposer",
                                "value": ""
                            },
                            "patternMatchUrlLoader": {
                                "type": "urlLoader",
                                "ref": "nominee_pincode",
                                "pattern": "^[1-9][0-9]{5}$",
                                "url": "https://some-pin-code-url/",
                                "queryParams": [["pincode"]],
                                "valueKey": "state#name",
                                "labelKey": "state#name",
                                "defaultValue": "0"
                            }
                        },
                        "events": {
                            "change": [
                                {
                                    "type": "setter",
                                    "ref": "nominee_city_name",
                                    "valueKey": "city#name"
                                },
                                {
                                    "type": "setter",
                                    "ref": "nominee_city",
                                    "valueKey": "city#id"
                                },
                                {
                                    "type": "setter",
                                    "ref": "nominee_state",
                                    "valueKey": "state#id"
                                }
                            ]
                        }
                    }
                }`

 While rendering the form first time, the API is failing and the form is throwing below error, I tried grasp from the previous implementation but couldn't.
 
<img width="959" alt="Screenshot 2024-05-06 at 8 07 17 PM" src="https://github.com/manojadams/mui-forms/assets/38013987/0b793dac-0a1b-4314-9980-26c083a23248">

**Comment by manojadams** [comment]
@venkateshpullaganti The schema you have attached here is partial.
Can you provide all the schema definition json.

**Reply by manojadams** [reply, marked as answer]
@venkateshpullaganti This issue has been fixed in the following version: 
_[^metaforms-core@4.0.3](https://www.npmjs.com/package/@manojadams/metaforms-core/v/4.0.3)_

And this can be accessed with the following version of mui-forms:
**[mui-forms@2.0.6](https://www.npmjs.com/package/mui-forms/v/2.0.6)**
